rero_ils/modules/utils.py

```diff
@@ -122,29 +122,76 @@ def get_record_class_update_permission_from_route(route_name):
             return record_class, update_permission
 
 
+def get_endpoint_configuration(module):
+    """Search into configuration file to find configuration for a module.
+
+    :param module: name of module (class name or endpoint name or module name)
+    :return: The configuration dictionary of the resource. 'None' if resource
+             is not found.
+    """
+    if not isinstance(module, str):
+        # Get the pid_type for the class
+        module = module.provider.pid_type
+    endpoints = current_app.config.get('RECORDS_REST_ENDPOINTS', {})
+    for idx, endpoint in endpoints.items():
+        search_index = endpoint.get('search_index')
+        if search_index == module or idx == module:
+            return endpoint
+
+
 def get_ref_for_pid(module, pid):
     """Get the $ref for a pid.
 
     :param module: name of module (class name or endpoint name or module name)
     :param pid: pid for record
     :return: url for record
     """
-    if not isinstance(module, str):
-        # Get the pid_type for the class
-        module = module.provider.pid_type
-    endpoints = current_app.config.get('RECORDS_REST_ENDPOINTS')
-    for endpoint in endpoints:
-        search_index = endpoints[endpoint].get('search_index')
-        # Try to find module in entpoints or entpoints.serch_index
-        if search_index == module or endpoint == module:
-            list_route = endpoints[endpoint].get('list_route')
-            if list_route:
-                return '{url}/api{route}{pid}'.format(
-                    url=current_app.config.get('RERO_ILS_APP_BASE_URL'),
-                    route=list_route,
-                    pid=pid
-                )
-    return None
+    configuration = get_endpoint_configuration(module)
+    if configuration and configuration.get('list_route'):
+        return '{url}/api{route}{pid}'.format(
+            url=current_app.config.get('RERO_ILS_APP_BASE_URL'),
+            route=configuration.get('list_route'),
+            pid=pid
+        )
+
+
+def extracted_data_from_ref(input, data='pid'):
+    """Extract a data from a `$ref` string.
+
+    :param input: string where to search data, or a dict containing '$ref' key
+    :param data: the data to found. Allowed values are :
+        * 'pid': the pid from the input
+        * 'resource': the resource search_index from input
+        * 'record_class': the record class to used to manage the input
+        * 'record': the record represented by the input
+
+    USAGE :
+      * extract_pid_from_ref('http://localhost/[resource]/[pid]', data='pid')
+      * extract_pid_from_ref({'$ref': 'http://localhost/[resource]/[pid]'})
+    """
+    def extract_part(input_string, idx=0):
+        parts = input_string.split('/')
+        if len(parts) > abs(idx):
+            return input_string.split('/')[idx]
+
+    if isinstance(input, str):
+        input = {'$ref': input}
+    if data == 'pid':
```

Comment:
If this code is used often or there are many elif this could be replaced by a dictonary:
https://jaxenter.com/implement-switch-case-statement-python-138315.html